homeassistant/components/octoprint/__init__.py

```diff
@@ -251,3 +253,17 @@ def device_info(self) -> DeviceInfo:
             name="OctoPrint",
             configuration_url=str(configuration_url),
         )
+
+
+async def async_migrate_entry(hass, config_entry: ConfigEntry):
```

Comment:
We should not migrate the entry version for this, as that makes downgrading impossible.

This can be migrated by updating the config entry during entry setup.